main.py

```diff
@@ -22,10 +23,19 @@
 class ApplicationWindow(QtWidgets.QMainWindow):
     """The main GUI window."""
 
+    # Define a new signal, used to add_frames from the seperate thread
+    add_frame = QtCore.pyqtSignal('PyQt_PyObject', 'PyQt_PyObject', 'PyQt_PyObject')
+    # Used when the add stream thread fails
+    fail_add_stream = QtCore.pyqtSignal('PyQt_PyObject', 'PyQt_PyObject')
```

Comment:
Is i possible to specify the objects preciser than `'PyQt_PyObject', 'PyQt_PyObject'`, if it is I think we should do that. If not, we'll act like it's raining.